### PR TITLE
Introduce start/end year for entity

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Introduce start/end year of period in which entity holidays are defined (with default values 1901 and 2100).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [x] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/thailand.py [hunk 1]
```diff
@@ -682,10 +680,10 @@ def _populate_workday_holidays(self):
 
         # วันลอยกระทง
         # Status: In-Use.
+        # Started in 1941.
```
Comment (PPsyrius) [inline holidays/countries/thailand.py:683]:
```suggestion
```
This is just a cut-off date due to the Thai LuniSolar calendar support range, not its start date.

**Review by PPsyrius — COMMENTED** [review]
I'm glad this is finally getting implemented ❤️ 

This might be something for later, but having subdivision-level start dates would be nice for countries such as Australia where each state has a different holiday start date the United States where the state level exists much earlier, but US federal holidays observance only started in [1870](https://en.wikipedia.org/wiki/Federal_holidays_in_the_United_States) - or a future #1852 implementation. :eyes:

File: holidays/countries/finland.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ class Finland(HolidayBase, ChristianHolidays, InternationalHolidays):
     default_language = "fi"
     supported_languages = ("en_US", "fi", "sv_FI", "uk")
     supported_categories = (PUBLIC, UNOFFICIAL)
+    start_year = 1853
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/finland.py:37]: Just value before the earliest appearance of holiday (Runeberg Day in 1854). 🙂

File: holidays/countries/thailand.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/thailand.py:683]: Yes, but I consider this attribute ( `start_year`) as just a technical, not as "start" in a juridical sense. In fact, "just a cut-off".

**Review by PPsyrius — APPROVED** [review]
LGTM 🕚

**Review by PPsyrius — APPROVED** [review]

**Pull request merged**